### ISSUE TITLE
Incorrect alignment if `def` return value is used

### ISSUE THREAD

**Issue by andrusha** [bbatsov/rubocop#2092]
Ruby 2.1 has `def` returning the method name as a symbol, which is useful for meta-programming:

```
irb(main):001:0> def cats_mew_mew
irb(main):002:1> end
=> :cats_mew_mew
```

An example use-case for this might be memoization:

```ruby
class Something
  include Memoize

  memoize def heavy_lifting(cat_food)
    something
  end
end
```

This seem to me as a correct formatting, however this raises two problems:

1. Automatic alignment would align `something` to `def` + spaces in the following way:

```ruby
memoize def heavy_lifting(cat_food)
          something
end
```

2. After alignement it'll raise error that `end` is not aligned with `def`:

```
W: end at 10, 2 is not aligned with def at 8, 10
  end
  ^^^
```

**Comment by bbatsov** [comment]
Can you check if that's a problem on `master`? If memory serves this behaviour was changed pretty recently.

**Comment by andrusha** [comment]
@bbatsov you're right, this is fixed in master. I'm closing the ticket.
